lib/job/arFileImportJob.class.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * Job worker for XML and CSV imports initiated from the WebUI.
+ *
+ * @package    symfony
+ * @subpackage jobs
+ */
+
+class arFileImportJob extends arBaseJob
+{
+  /**
+   * @see arBaseJob::$requiredParameters
+   */
+  public function runJob($parameters)
+  {
+    $this->info($this->i18n->__('Importing %1 file: %2.', array('%1' => strtoupper($parameters['importType']), '%2' => $parameters['file']['name'])));
+
+    // set indexing preference
+    if (true == $parameters['noIndex'])
+    {
+      QubitSearch::disable();
+    }
+
+    try
+    {
+      switch ($parameters['importType'])
+      {
+        case 'csv':
+          $importer = new QubitCsvImport;
+          if ($parameters['doCsvTransform']) $this->info($this->i18n->__('Applying transformation to CSV file.'));
+          $importer->doCsvTransform = $parameters['doCsvTransform'];
+
+          if ($parameters['noIndex']) $this->info($this->i18n->__('Indexing imported records.'));
```

Comment:
We as a rule always use curly braces even when it's not strictly necessary with the 1-liners like this.